### DISCUSSION TITLE
Import *.rrd from https://github.com/wiedehopf/graphs1090#graphs1090

### DISCUSSION THREAD

**Discussion by dranreb38** [jprochazka/adsb-receiver#604] [Q&A]
I would like to keep all my data collected by the script of https://github.com/wiedehopf/graphs1090#graphs1090 and then import it into your project. I tried to copy the rdd folder into your installation (/var/lib/collectd/rrd/localhost) but without success! Do you have any idea how to import *.rrd files

**Comment by jprochazka** [comment]
[Graphs1090](https://github.com/wiedehopf/graphs1090#graphs1090) would in fact appear to use a different data format than this project as outlined [here](https://github.com/wiedehopf/graphs1090?tab=readme-ov-file#resetting-the-database-format)? 

However, being that @wiedehopf's graphs1090 is a much more up to date version of dump1090-tools/ADS-B Portal graphs. I would recommend sticking with his graphs for the time being. I will go ahead and create a ticket geared towards adding graphs1090 support to the install scripts. Adding an option to install graphs1090 instead of using the existing graphs system should be a trivial task and one I am willing to undertake over continued support for a system I am looking to replace in v3.x*.

I am currently testing the next release which will include VDL Mode 2 support and should be able to squeeze in support for graph1090 for this release as well. How does a release adding said support sometime mid to late this week sound?

_* As mentioned in previous announcements one of my goals with v3.x is to replace the dated Collectd/RRDTools graphs currently implemented within this project with a more modern/versatile solution. When implemented I will be sure to add proper data conversion options for both the current ADS-B Portal rrd files as well as those generated by graphs1090._

**Reply by wiedehopf** [reply]
graphs1090 is still collectd / rrd.
the difference in DB format is mostly in relation to things like range limited to 590 / message rate limit values people were hitting in weird usecases.
So most those limits are gone. That required such a DB format reset.
Similar for when people asked for longer time graphs, there is now graphs for up to 10 yrs.

There is no good replacement for rrd, it all hogs lots of storage / doesn't seem as robust.

Not all the data will be there, but you can change from the mutability graphs install to graphs1090 and it'll carry over most of the existing data.
And it'll work without resetting the format, but it will be limited to 2 yrs and some other limitations.

**Comment by jprochazka** [comment]
Just an update, I started working on offering the ability to install Readsb as a decoder option so that a graphs1090 installation option would be fully functional as far as displaying data is concerned which wiedenhopf pointed out would be an issue. I ran into a problem however in that Readsb does not play nice with Ubuntu with the recommendation to use Debian or Docker images as the solution. Both options go against the goals of this project so for the time being I am going to have to shelve this solution.

> I tried to copy the rdd folder into your installation (/var/lib/collectd/rrd/localhost) but without success!

> the difference in DB format is mostly in relation to things like range limited to 590 / message rate limit values people were hitting in weird usecases.

If in fact the only changes to the rrd databases are size constraints...

@dranreb38 what does the file structure look like? It should look something like this.

/var/lib/collectd/rrd/localhost/aggregation-cpu-average
/var/lib/collectd/rrd/localhost/df-root
/var/lib/collectd/rrd/localhost/disk-mmcblk0
/var/lib/collectd/rrd/localhost/disk-mmcblk0
/var/lib/collectd/rrd/localhost/interface-eth0
/var/lib/collectd/rrd/localhost/memory
/var/lib/collectd/rrd/localhost/thermal-thermal_zone0
/var/lib/collectd/rrd/localhost/thermal-thermal_zone0
/var/lib/collectd/rrd/localhost/interface-wlan0
/var/lib/collectd/rrd/localhost/dump1090-localhost
/var/lib/collectd/rrd/localhost/dump1090-localhost
/var/lib/collectd/rrd/localhost/dump1090-localhost
/var/lib/collectd/rrd/localhost/dump1090-localhost
/var/lib/collectd/rrd/localhost/dump1090-localhost
/var/lib/collectd/rrd/localhost/dump1090-localhost

> There is no good replacement for rrd, it all hogs lots of storage / doesn't seem as robust.

I am leaning towards replacing rrdtools with [Prometheus](https://prometheus.io/) in the new portal. I am planning on moving graph generation from the backend/receiver to the frontend/client side so the rrdtools package will lose what little value add it brought to the table as far as the portal will be concerned. Tools such as Grafana can connect to Prometheus as a data source fulfilling a long-standing request for a customizable graphing solution.

**Comment by wiedehopf** [comment]
graphs1090 works fine with dump1090-fa

If you do an automated build for Ubuntu 24 .. just do
```
sed -i -e 's/librtlsdr0/librtlsdr2/g' debian/control
```
before building.
problem solved.

**Reply by jprochazka** [reply]
Thanks for the information and all the work you have done on your projects. I added the sed command to the install script and added install options for Graphs1090 and tar1090 as well. I need to test/debug them but unless something unforeseen happens the option to install these items should be made available along with vdlm2 support sometime this weekend.

**Reply by wiedehopf** [reply]
This should no longer be needed due to a PR to the control file, apparently you can simply use | to use one package or another.

**Reply by jprochazka** [reply]
Did not know that but is good to know for occasions such as this.

**Reply by wiedehopf** [reply]
I don't have Ubuntu24 on hand but maybe you could test?

https://github.com/wiedehopf/readsb/pull/77

It's already in dev, there is also the stale branch which i use for https://github.com/wiedehopf/adsb-scripts/wiki/Automatic-installation-for-readsb

But either branch should be fine, it's very rare for problematic versions to be pushed to dev :)

**Reply by jprochazka** [reply]
I sure can. Actually, about to test it out now. Will report back shortly.

**Reply by jprochazka** [reply]
I am away from home at the moment so I do not have access to an RTL-SDR dongle at the moment but I can confirm that Readsb is in fact building on both Ubuntu Noble Numbat and Debian Bookworm. There is one thing missing from your build instructions and that is at `sudo apt install --no-install-recommends --no-install-suggests -y ...` in the README.md file you might want to add the fakeroot package.

Note: I did a manual build using the install script located in the vdlm2 branch of this project.

**Comment by dranreb38** [comment]
@jprochazka , yes 
![Screenshot_555](https://github.com/user-attachments/assets/5f24507d-432e-44d7-96f3-d644eed238a6)
![Screenshot_556](https://github.com/user-attachments/assets/19d3da00-16c0-4e37-b1c7-30c4d3bf980e)

**Reply by dranreb38** [reply]
Do I need to run a command to integrate the *.rrd files? I also did an export in xml

**Reply by jprochazka** [reply]
Everything looks proper as far as paths are concerned.
Do you get any errors when running the graph generation command we could use to further investigate things?

`bash ~/adab-receiver/build/graphs/make-collectd-graphs.sh 1h`
 
If you exported the data as XML you will need to use the rrdrestore command in order to create proper rrd databases.
https://oss.oetiker.ch/rrdtool/doc/rrdrestore.en.html

**Comment by dranreb38** [comment]
@jprochazka 
I transferred to your project all the *.rrd files from my "old RPI" in the folder /var/lib/collectd/rrd/localhost/dump1090-localhost in the new one. Then launched the script "make-collectd-graphs.sh 1h" but I don't have the data from my old RPI yet, maybe I have to wait a little? ......
there are errors but it's normal because I work with an SSD and not an SD card.

![Screenshot_558](https://github.com/user-attachments/assets/08b38588-3a54-4409-8619-f1e578f070bf)

For 30 Days make-collectd-graphs.sh 30d ???

**Reply by jprochazka** [reply]
I do not see any errors other than the one you explained. 

Are files being generated and saved to the graphs folder in Lighttpd's document root which is by default `/var/www/html/graphs`? If they do exist how do they look? Do they display any data? You can get Lighttpd's document root using this command.

`/usr/sbin/lighttpd -f /etc/lighttpd/lighttpd.conf -p | grep server.document-root`

If you want to graph disk usage for your SSD change `disk-mmcblk0` to `disk-sda` at lines 628 and 629 of `~/adsb-receiver/build/portal/make-collectd-graphs.sh` and in `/etc/collectd/collectd.conf` look for the following code block and swap `mmcblk0` with `sda`.

```
<Plugin "disk">
	Disk "mmcblk0"
	IgnoreSelected false
</Plugin>
```
Offering the option to change this more easily has been something I have been meaning to do. 

> For 30 Days make-collectd-graphs.sh 30d ???

Yes, that will generate 30-day graphs.

**BTW**

The next version containing a Graphs1090 option should be finalized in the vdlm2 branch and should be released sometime today.

**Reply by jprochazka** [reply]
> Adding an option to install graphs1090 instead of using the existing graphs system should be a trivial task

I merged the vdlm2 branch into the master branch tonight which adds an option to install Graphs1090 in case you would rather stick to that. I did not modify the PHP portal so you will want to disable the default graphs link and create a custom link to the Graphs1090 page in the PHP portal administration area.

**Reply by jprochazka** [reply]
If you are in fact still having issues and do not mind sharing your rrd data with me, I would be willing to take a shot at it if you can get me a copy of said rrd data. Let me know if you would like to go that route so we can arrange a way to share said data if you are not comfortable sharing it here.

**Reply by dranreb38** [reply]
@jprochazka  
Send url for download data my history 
Regards

**Reply by jprochazka** [reply]
Sorry for the delay getting back to you.  I have been settling into a new job this week. I wish they had a DM feature oh well I will leave this link for you to use. Hopefully it won't get abused.

https://www.dropbox.com/request/drTxGqi5QcV5XP4Iqxg5

**Reply by dranreb38** [reply]
Hello Joseph, thanks for the link I upload my data files in *.rrd format from my feeder (dranreb38 / T-LFH7) Thank you for your work!
Kind regards

**Comment by dranreb38** [comment]
Hello Joseph, 
have you found a way to integrate the history of the graphs ?
Kind regards

**Reply by jprochazka** [reply]
I am so sorry. I saw that the question was answered and thought you resolved the issue. I will begin looking into it today. I went ahead and unmarked the question as answered until we get the issue resolved.
